### PR TITLE
Dialog Border Radius Migration Guide

### PR DESCRIPTION
**Pull request by Piinks**
Migration guide for: ~https://github.com/flutter/flutter/pull/56084~
https://github.com/flutter/flutter/pull/58829

Related Issues
https://github.com/flutter/flutter/pull/58839
https://github.com/flutter/flutter/pull/58841

### CODE REVIEW

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: Dialogs' Default BorderRadius
+description: The default BorderRadius of Dialog widgets is changing.
+---
+
+## Summary
+
+Instances of `Dialog`, as well as `SimpleDialog`, `AlertDialog`, and `showTimePicker`, now has
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:8]: has->have

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+a default shape of a `RoundedRectangleBorder` with a `BorderRadius` of 4.0 pixels. This matches the
+current specifications of Material Design. Prior to this change, the default behavior for
+`Dialog.shape`'s `BorderRadius` was 2.0 pixels.
+
+## Context
+
+Use of `Dialog`s and their associated subclasses (`SimpleDialog`, `AlertDialog`, and
+`showTimePicker`), appears slightly different as the border radius is larger. Golden file tests
+will fail if they contain a Dialog` in the new default behavior.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:17]: missing \`

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+This change was initially introduced with an opt-in flag
+`useMaterialBorderRadius`, to allow developers time to migrate
+without experiencing broken code. This temporary opt-in flag will be removed in a follow up change.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:21]: this paragraph will be moot once anyone cares about this guide, so we should remove it

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+The `showDatePicker` dialog already matches this specification and is unaffected by this change.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:23]: "already matched"

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+## Migration guide
+
+By applying the opt-in flag, `useMaterialBorder`, you can check the layout of your `Dialog` widgets
+and check against exiting tests to see if they need to be updated.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:28]: we should assume the flag is gone unless we're actually planning on supporting it for several quarters.

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            );
+          }
+        ),
+      ),
+    );
+  }
+}
+```
+
+Code to validate migration:
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:86]: (as per above, this one should presumably go away)

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        ),
+      ),
+    );
+  }
+}
+```
+
+## Timeline
+
+TBD - Add the version number that contains this change.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:118]: do this before landing

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+`showTimePicker`), appears slightly different as the border radius is larger. Golden file tests
+will fail if they contain a Dialog` in the new default behavior.
+
+This change was initially introduced with an opt-in flag
+`useMaterialBorderRadius`, to allow developers time to migrate
+without experiencing broken code. This temporary opt-in flag will be removed in a follow up change.
+
+The `showDatePicker` dialog already matches this specification and is unaffected by this change.
+
+## Migration guide
```
Comment (Hixie) [inline src/docs/release/breaking-changes/dialog-border-radius.md:25]: sounds like really there's no migration unless you want to keep the old behavior or unless you have goldens; i would rephrase this section to be more in line with those use cases.

**Comment by Piinks** [comment]
Thanks for the feedback! I understand now that this migration guide should reflect the final intended behavior. Since there are a few more changes to make before then, I am going to close this. I'll re-open when it's ready. :)

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+## Summary
+
+Instances of `Dialog`, as well as `SimpleDialog`, `AlertDialog`, and `showTimePicker`, now have
+a default shape of a `RoundedRectangleBorder` with a `BorderRadius` of 4.0 pixels. This matches the
+current specifications of Material Design. Prior to this change, the default behavior for
+`Dialog.shape`'s `BorderRadius` was 2.0 pixels.
+
+## Context
+
+Use of `Dialog`s and their associated subclasses (`SimpleDialog`, `AlertDialog`, and
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/dialog-border-radius.md:15]: Delete "Use of".

File: src/docs/release/breaking-changes/dialog-border-radius.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+`showTimePicker`), appears slightly different as the border radius is larger. Golden file tests
+will fail if they contain a `Dialog` in the new default behavior.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/dialog-border-radius.md:17]: Am I reading this right? Golden tests will fail if the Dialog now uses a 4-pixel border? And the only way to fix this is to force the dialog to use a 2-pixel border?

Comment (Piinks) [reply src/docs/release/breaking-changes/dialog-border-radius.md]: I've re-worded it. Please let me know if it is clearer.
A golden file test is a picture of what a widget is supposed to look like. I have changed what it looks like, so either the code should be updated to make it match the image again, or the image should be updated.
